awx_collection/plugins/modules/tower_job_template.py

```diff
@@ -12,410 +13,430 @@
+        }
+    })
+
+    # Create the data that gets sent for create and update
+    new_fields = {}
+    if scm_branch:
+        new_fields['scm_branch'] = scm_branch
+    if ask_tags_on_launch:
+        new_fields['ask_tags_on_launch'] = ask_tags_on_launch
+    if job_type:
+        new_fields['job_type'] = job_type
+    if skip_tags:
+        new_fields['skip_tags'] = skip_tags
+    if ask_verbosity_on_launch:
+        new_fields['ask_verbosity_on_launch'] = ask_verbosity_on_launch
+    if playbook:
+        new_fields['playbook'] = playbook
+    if survey_enabled:
+        new_fields['survey_enabled'] = survey_enabled
+    if description:
+        new_fields['description'] = description
+    if ask_diff_mode_on_launch:
+        new_fields['ask_diff_mode_on_launch'] = ask_diff_mode_on_launch
+    if job_tags:
+        new_fields['job_tags'] = job_tags
+    if allow_simultaneous:
+        new_fields['allow_simultaneous'] = allow_simultaneous
+    if diff_mode:
+        new_fields['diff_mode'] = diff_mode
+    if ask_inventory_on_launch:
+        new_fields['ask_inventory_on_launch'] = ask_inventory_on_launch
+    if inventory:
+        new_fields['inventory'] = inventory_id
+    if limit:
+        new_fields['limit'] = limit
+    if forks:
+        new_fields['forks'] = forks
+    if become_enabled:
+        new_fields['become_enabled'] = become_enabled
+    if force_handlers:
+        new_fields['force_handlers'] = force_handlers
+    if ask_variables_on_launch:
+        new_fields['ask_variables_on_launch'] = ask_variables_on_launch
+    if ask_job_type_on_launch:
+        new_fields['ask_job_type_on_launch'] = ask_job_type_on_launch
+    if start_at_task:
+        new_fields['start_at_task'] = start_at_task
+    if ask_limit_on_launch:
+        new_fields['ask_limit_on_launch'] = ask_limit_on_launch
+    if webhook_credential:
+        new_fields['webhook_credential'] = webhook_credential_id
+    if custom_virtualenv:
+        new_fields['custom_virtualenv'] = custom_virtualenv
+    if host_config_key:
+        new_fields['host_config_key'] = host_config_key
+    if job_slice_count:
+        new_fields['job_slice_count'] = job_slice_count
+    if ask_skip_tags_on_launch:
+        new_fields['ask_skip_tags_on_launch'] = ask_skip_tags_on_launch
+    new_fields['name'] = new_name if new_name else name
+    if use_fact_cache:
+        new_fields['use_fact_cache'] = use_fact_cache
+    if extra_vars:
+        new_fields['extra_vars'] = extra_vars
+    if verbosity:
+        new_fields['verbosity'] = verbosity
+    if ask_credential_on_launch:
+        new_fields['ask_credential_on_launch'] = ask_credential_on_launch
+    if project:
+        new_fields['project'] = project_id
+    if webhook_service:
+        new_fields['webhook_service'] = webhook_service
+    if timeout:
+        new_fields['timeout'] = timeout
+    if ask_scm_branch_on_launch:
+        new_fields['ask_scm_branch_on_launch'] = ask_scm_branch_on_launch
```

Comment:
This really goes past the point where I would rather put them in a loop

Reply:
I'll take a look at the auto-generation template and see if it would be easy enough to create a loop here rather than the individual lines.